src/hoc/BadgerBase/BadgerBase.js

```diff
@@ -283,6 +308,56 @@ const BadgerBase = (Wrapped: React.AbstractComponent<any>) => {
 			this.setState({ intervalUnconfirmed: intervalUnconfirmedNext });
 		};
 
+		setupWatchInvoice = async () => {
+			const { paymentRequestUrl } = this.props;
+
+			this.ws = new WebSocket(
+				`wss://pay.bitcoin.com/s/${paymentRequestUrl.slice(26)}`
```

Comment:
Let's properly get the payment ID.  
A basic way could be something such as... 

```javascript
const urlParts = paymentRequestURL.split('/');
const paymentId = urlParts[-1]
this.ws = new WebSocket(`wss://pay.bitcoin.com/s/${paymentId}`)
```
There should also be a check that this is a `pay.bitcoin` invoice before running this code so it doesn't just break for other implementations.

Reply:
6431c1c